util/indexer-r/src/indexer/mod.rs

```diff
@@ -0,0 +1,258 @@
+                block.number() as i64,
+                &mut tx,
+            )
+            .await?;
+        }
+        tx.commit().await.map_err(|err| Error::DB(err.to_string()))
+    }
+
+    pub(crate) async fn rollback(&self) -> Result<(), Error> {
+        let indexer_handle = AsyncIndexerRHandle::new(self.store.clone(), self.pool.clone());
+        let tip = indexer_handle
+            .query_indexer_tip()
+            .await
+            .map_err(|err| Error::DB(err.to_string()))?;
+
+        if let Some(tip) = tip {
+            let mut tx = self
+                .store
+                .transaction()
+                .await
+                .map_err(|err| Error::DB(err.to_string()))?;
+
+            rollback_block(tip.block_hash, &mut tx).await?;
+
+            return tx.commit().await.map_err(|err| Error::DB(err.to_string()));
+        }
+
+        Ok(())
+    }
+
+    pub(crate) async fn insert_transactions(
+        &self,
+        block_view: &BlockView,
+        tx: &mut Transaction<'_, Any>,
+    ) -> Result<(), Error> {
+        let block_hash = block_view.hash().raw_data().to_vec();
+        let tx_views = block_view.transactions();
+        let mut matched_tx_hashes = HashSet::new();
+        let mut output_cell_rows = Vec::new();
+        let mut input_rows = Vec::new();
+        let mut output_association_script_rows = Vec::new();
+        let mut script_set = HashSet::new();
+
+        for tx_view in tx_views.iter() {
+            for ((output_index, (cell, data)), out_point) in tx_view
+                .outputs_with_data_iter()
+                .enumerate()
+                .zip(tx_view.output_pts_iter())
+            {
+                if self
+                    .custom_filters
+                    .is_cell_filter_match(&cell, &data.pack())
+                {
+                    build_output_cell_rows(
+                        &cell,
+                        tx_view,
+                        output_index,
+                        &data,
+                        &mut output_cell_rows,
+                    )?;
+                    build_script_set(&cell, &mut script_set, tx).await?;
+                    build_output_association_script_rows(
+                        &cell,
+                        &out_point,
+                        &mut output_association_script_rows,
+                    )?;
+                    matched_tx_hashes.insert(tx_view.hash());
+                }
+            }
+        }
+        bulk_insert_output_table(&output_cell_rows, tx).await?;
+        bulk_insert_script_table(&script_set, tx).await?;
+
+        // The output needs to be inserted into the db-transaction before the input traversal.
+        // This is to cope with the case where the output is spent in a transaction in the same block,
+        // because the input needs to query the corresponding output cell when doing a cell filter.
+        for (tx_index, tx_view) in tx_views.iter().enumerate() {
```

Comment:
How about use `.skip(1)` to skip cellbase tx?

Reply:
yes, it looks cleaner with skip. I'll change it to something like this:

```rust
// Skip the first cellbase transaction.
for tx_view in tx_views.iter().skip(1) {
...
}
```